### PR TITLE
[Express Route] IPv6 Global Reach Add/Set-AzExpressRouteCircuitConnection cmdlet changes.

### PR DESCRIPTION
**Pull request by amwate**
## Description
We are in the process of adding support for IPv6 in ExpressRouteCircuitConnectionConfig.

In the process, we have updated the existing PowerShell cmdlet 
Add-AzExpressRouteCircuitConnectionConfig to accept another optional parameter AddressPrefixType (that optionally specifies the address family type for the address prefix). 

In addition, we also introduced another cmdlet to modify the existing ExpressRouteCircuitConnectionConfig to add a new block IPv6CircuitConnectionConfig as well. 
This would be used to update any existing GlobalReach configuration (ExpressRouteCircuitConnection) configurations.

Below is the design PR
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/451#event-2913187208

This pull request brings in the changes in the powershell sdk. 
The relevant swagger changes are to be merged in network-december-release.
https://github.com/Azure/azure-rest-api-specs/pull/8145

NRP changes are already merged. 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by VeryEarly** [comment]
Hi @amwate ,

there are build failures due to 
"Common\NetworkResourceManagerProfile.cs(750,92): error CS0234: The type or namespace name 'Ipv6CircuitConnectionConfig' does not exist in the namespace 'Microsoft.Azure.Management.Network.Models' (are you missing an assembly reference?)"

cannot find Ipv6CircuitConnectionConfig in
[Microsoft.Azure.Management.Network_19.17.0-preview](https://github.com/Azure/azure-sdk-for-net/tree/Microsoft.Azure.Management.Network_19.17.0-preview/sdk/network/Microsoft.Azure.Management.Network/src/Generated/Models)

File: src/Network/Network/ExpressRouteCircuit/Peering/Connection/SetAzureExpressRouteCircuitConnectionConfigCommand.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        {
+            base.Execute();
+
+            var peering = this.ExpressRouteCircuit.Peerings.SingleOrDefault(
+                    resource =>
+                        string.Equals(resource.Name, "AzurePrivatePeering", System.StringComparison.CurrentCultureIgnoreCase));
+
+            if (peering == null)
+            {
+                throw new ArgumentException("Private Peering needs to be configured on the Express Route Circuit");
```
Comment (isra-fel) [inline src/Network/Network/ExpressRouteCircuit/Peering/Connection/SetAzureExpressRouteCircuitConnectionConfigCommand.cs]: Please use resource string.

File: src/Network/Network/ExpressRouteCircuit/Peering/Connection/SetAzureExpressRouteCircuitConnectionConfigCommand.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            }
+
+            var circuitconnection = peering.Connections.SingleOrDefault(
+                    resource =>
+                        string.Equals(resource.Name, Name, StringComparison.CurrentCultureIgnoreCase));
+
+            if (null == circuitconnection)
+            {
+                throw new ArgumentException(string.Format("Circuit Connection with Name {0} was not added to the private peering", Name));
```
Comment (isra-fel) [inline src/Network/Network/ExpressRouteCircuit/Peering/Connection/SetAzureExpressRouteCircuitConnectionConfigCommand.cs]: Please use resource string.

File: src/Network/Network/ExpressRouteCircuit/Peering/Connection/SetAzureExpressRouteCircuitConnectionConfigCommand.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.WindowsAzure.Commands.Common.CustomAttributes;
+using System;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network {
+    [CmdletOutputBreakingChange(typeof(PSExpressRouteCircuit), DeprecatedOutputProperties = new[] { "AllowGlobalReach" })]
```
Comment (isra-fel) [inline src/Network/Network/ExpressRouteCircuit/Peering/Connection/SetAzureExpressRouteCircuitConnectionConfigCommand.cs]: Why will there be a breaking change?

**Comment by amwate** [comment]
Creating a new pull request https://github.com/Azure/azure-powershell/pull/11501.

**Pull request closed without merging**